wys/api/python/wys_api.py

```diff
@@ -357,45 +401,59 @@ def update_locations(conn, loc_table):
+                loc text,
+                geom geometry);
         """
         cur.execute(create_temp_table)
         execute_values(cur, 'INSERT INTO daily_intersections (api_id, address, sign_name, dir, start_date, loc) VALUES %s', loc_table)
+        calc_geom_temp_table = """
+            UPDATE daily_intersections
+            SET geom = st_setsrid(st_makepoint(split_part(regexp_replace(loc, '[()]'::text, ''::text, 'g'::text), ','::text, 2)::double precision, 
+                                               split_part(regexp_replace(loc, '[()]'::text, ''::text, 'g'::text), ','::text, 1)::double precision), 
+                                  4326);
+            """
+        cur.execute(calc_geom_temp_table)
         update_locations_sql="""
             WITH locations AS (
-                SELECT api_id, address, sign_name, dir, loc, max(start_date)
+                SELECT DISTINCT ON(api_id) api_id, address, sign_name, dir, loc, start_date, geom
                 FROM wys.locations 
-                GROUP BY api_id, address, sign_name, loc, dir
+                ORDER BY api_id, start_date DESC
             ),
             differences AS (
-                SELECT a.api_id, a.address, a.sign_name, a.dir, start_date, 
-                       a.loc 
+                SELECT a.api_id, a.address, a.sign_name, a.dir, a.start_date, 
+                       a.loc, a.geom 
                 FROM daily_intersections A
                 LEFT JOIN locations B ON A.api_id = B.api_id
                                       AND A.sign_name = B.sign_name 
-                WHERE B.sign_name IS NULL 
+                WHERE B.sign_name IS NULL
+                      AND (st_distance_sphere(A.geom, B.geom) > 100 
+                           OR A.dir <> B.dir)
                 
                 UNION
                 
-                SELECT a.api_id, a.address, a.sign_name, a.dir, start_date,     
-                       a.loc 
+                SELECT a.api_id, a.address, a.sign_name, a.dir, a.start_date,     
+                       a.loc, a.geom 
                 FROM daily_intersections A
                 LEFT JOIN locations B ON A.api_id = B.api_id
                                       AND B.address = A.address
                 WHERE B.address IS NULL
+                      AND (st_distance_sphere(A.geom, B.geom) > 100 
+                           OR A.dir <> B.dir)
                 
                 UNION 
                 
-                SELECT a.api_id, a.address, a.sign_name, a.dir, start_date, 
-                       a.loc
+                SELECT a.api_id, a.address, a.sign_name, a.dir, a.start_date, 
+                       a.loc, a.geom
                 FROM daily_intersections A
                 LEFT JOIN locations B ON A.api_id = B.api_id
                                       AND B.loc = A.loc
                 WHERE B.loc IS NULL
+                      AND (st_distance_sphere(A.geom, B.geom) > 100 
+                           OR A.dir <> B.dir)
```

Comment:
I don't think this distance comparison will happen because of the anti-join. 
Would need to be 

```sql
LEFT JOIN locations B ON A.api_id = B.api_id
                                     
                WHERE B.loc <> A.loc
                      AND (st_distance_sphere(A.geom, B.geom) > 100 
                           OR A.dir <> B.dir)

Reply:
By monthly functions, do you mean `update_mobile_api_id` and `update_stationary_signs`?

Reply:
More missing dates to pull after merge:
* 2022-03-29
* 2022-03-31